core/domain/caching_services.py

```diff
@@ -0,0 +1,221 @@
+        sub-namespace.
+    """
+    if not sub_namespace:
+        return '%s-%s' % (namespace, obj_id)
+    else:
+        return '%s-%s-%s' % (namespace, sub_namespace, obj_id)
+
+
+def flush_memory_cache():
+    """Flushes the memory cache by wiping all of the data."""
+    memory_cache_services.flush_cache()
+
+
+def get_multi(namespace, sub_namespace, obj_ids):
+    """Get a dictionary of the {key, value} pairs from the memory cache.
+
+    Args:
+        namespace: str. The namespace under which the values associated with
+            these object ids lie. The namespace determines how the keys are
+            decoded from their JSON encoded string. Use CACHE_NAMESPACE_DEFAULT
+            as namespace for objects that don't require deserialization.
+        sub_namespace: str|None. Sub namespace further differentiates the
+            values. For Explorations, Skills, Stories, Topics, Collections, the
+            sub_namespace is the version number of the objects. If sub-namespace
+            is not required, pass in None.
+        obj_ids: list(str). List of object ids corresponding to values to
+            retrieve from the cache.
+
+    Raises:
+        ValueError. The namespace does not exist or is not recognized.
+
+    Returns:
+        dict(str, Exploration|Skill|Story|Topic|Collection|str). Dictionary of
+        decoded (key, value) pairs retrieved from the platform caching service.
+    """
+    result_dict = {}
+    if len(obj_ids) == 0:
+        return result_dict
+
+    if namespace not in DESERIALIZATION_FUNCTIONS:
+        raise ValueError(
+            'Invalid namespace: %s.' % namespace)
+
+    namespaced_keys = []
+    for obj_id in obj_ids:
+        namespaced_keys.append(
+            _get_namespaced_key_from_type(obj_id, namespace, sub_namespace))
+    values = memory_cache_services.get_multi(namespaced_keys)
+    for obj_id, value in python_utils.ZIP(obj_ids, values):
+        if value:
+            deserialization_function = (
+                _get_deserialization_function_for_namespace(namespace))
+            result_dict[obj_id] = deserialization_function(value)
+
+    return result_dict
+
+
+def set_multi(namespace, sub_namespace, id_value_mapping):
+    """Set multiple id' values at once to the cache where the values are all
+    of a specific type or a primitype type.
+
+    Args:
+        namespace: str. The namespace under which the values associated with the
+            id lies. Use CACHE_NAMESPACE_DEFAULT as namespace for objects that
+            don't require serialization.
+        sub_namespace: str|None. Sub namespace further differentiates the
+            values. For Explorations, Skills, Stories, Topics, Collections, the
+            sub_namespace is the version number of the objects. If sub-namespace
+            is not required, pass in None.
+        id_value_mapping: dict(str, *). A dict of {id, value} pairs to set to
```

Comment:
Here/elsewhere: change * to Exploration|Collection|... (i.e. specify the types explicitly)